logstash-core/src/main/java/co/elastic/logstash/api/Filter.java

```diff
@@ -53,6 +53,8 @@ public interface Filter extends Plugin {
      */
     Collection<Event> filter(Collection<Event> events, FilterMatchListener matchListener);
 
+    void close();
```

Comment:
If we add a _default_ method to our interface, we can avoid breaking plugins that were written against a version of this interface that didn't have the method.

```suggestion
    default void close() { return; };
```

Reply:
+1 for making the method default. Otherwise we may loose backward compatibility.
In [Input](https://github.com/mashhurs/logstash/blob/a956e2e499631c412be3399390f3258aaef0b891/logstash-core/src/main/java/co/elastic/logstash/api/Input.java#L54) and [Output](https://github.com/mashhurs/logstash/blob/a956e2e499631c412be3399390f3258aaef0b891/logstash-core/src/main/java/co/elastic/logstash/api/Output.java#L42) we named it `stop`, I think we should standardize... maybe moving it to Plugin level  would also make sense to provide same behaviour to all plugins.

Reply:
Yes that's a good idea.